### ISSUE TITLE
ProvisionedThroughputExceededException on Retry

### ISSUE THREAD

**Issue by jagilbert27** [awslabs/aws-lambda-redshift-loader#177]
When something fails (not sure why yet) The retry seems to hammer so hard on the table that I get ProvisionedThroughputExceededException.

`START RequestId: ef2d94fb-f3fc-11e8-bfe9-0161362fa823 Version: $LATEST

17:36:43 2018-11-29T17:36:43.439Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Found Redshift Load Configuration for usan-cdrlambda-stage/ctg/detail

17:36:43 2018-11-29T17:36:43.469Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Adding Pending Batch Entry for usan-cdrlambda-stage/ctg/detail/ctg_redshift_detail_550047864.csv

17:36:43 2018-11-29T17:36:43.647Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Batch 22a83c38-a84b-483a-b26b-15a5a975e2cf still current after configuration reload attempt 1. Recycling in 0 ms.

17:36:43 2018-11-29T17:36:43.846Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Batch 22a83c38-a84b-483a-b26b-15a5a975e2cf still current after configuration reload attempt 2. Recycling in 0 ms.

17:36:44 2018-11-29T17:36:44.439Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Batch 22a83c38-a84b-483a-b26b-15a5a975e2cf still current after configuration reload attempt 3. Recycling in 0 ms.

17:36:45 2018-11-29T17:36:45.108Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Batch 22a83c38-a84b-483a-b26b-15a5a975e2cf still current after configuration reload attempt 4. Recycling in 0 ms.

17:36:45 2018-11-29T17:36:45.257Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Batch 22a83c38-a84b-483a-b26b-15a5a975e2cf still current after configuration reload attempt 5. Recycling in 0 ms.

17:36:45 2018-11-29T17:36:45.438Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	Batch 22a83c38-a84b-483a-b26b-15a5a975e2cf still current after configuration reload attempt 6. Recycling in 0 ms.

17:37:11 2018-11-29T17:37:11.030Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	{ ProvisionedThroughputExceededException: The level of configured provisioned throughput for the table was exceeded. Consider increasing your provisioning level with the UpdateTable API. at Request.extractError (/var/runtime/node_modules/aws-sdk/lib/protocol/json.js:48:27) at Request.callListeners (/var/runtime/node_modules/aws-

17:37:11 2018-11-29T17:37:11.039Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	{"message":"The level of configured provisioned throughput for the table was exceeded. Consider increasing your provisioning level with the UpdateTable API.","code":"ProvisionedThroughputExceededException","time":"2018-11-29T17:37:11.030Z","requestId":"E3VENR60QK2IHO1BN9TRMF6RTBVV4KQNSO5AEMVJF66Q9ASUAAJG","statusCode":400,"retryable":tr

17:37:11 2018-11-29T17:37:11.039Z	ef2d94fb-f3fc-11e8-bfe9-0161362fa823	{"errorMessage":"error"}

17:37:11 END RequestId: ef2d94fb-f3fc-11e8-bfe9-0161362fa823`

**Comment by jagilbert27** [comment]
Is the ProvisionedThroughputExceededException referring to Redshift or Dynamo?

**Comment by jagilbert27** [comment]
The concern is that the traffic goes through the roof with all the retries after 0ms.

**Comment by IanMeyers** [comment]
Yeah that's clearly an issue with the retry not moving off a 0 ms duration. Looking.

**Comment by jagilbert27** [comment]
Thanks Ian.
Still trying to figure out the problem that causes the error/retry.
None of the LambdaRedshiftBatches have a status=locked.
I  have 11 LambdaRedshiftBatchLoadConfig items.  Only 2 of them cause the error/retry...
Any troubleshooting ideas would be greatly appreciated.
Cheers,
Jeff

**Comment by IanMeyers** [comment]
Definitely need to get to the logs. Are you able to get to the specific invocations against a given config?

Thx,

Ian

**Comment by jagilbert27** [comment]
Thanks Ian,
I deleted the two offending LambdaRedshiftBatchLoadConfig entries, and all the LambdaRedshiftBatches associated with them, then recreated the LambdaRedshiftBatchLoadConfig items.  That seems to have done the trick.  So I am ok now. (yay)

As for logs, I have the cloudwatch logs, the failed manafests and the SNS Notifications.
They all contain something like this:
```
"Unable to write usan-cdrlambda-stage/ctg/header/ctg_redshift_header_790047888.csv in 100 attempts.
Failing further processing to Batch 16f031c1-514f-407d-8a38-e1440acb1613 which may be stuck in 'locked' state. 
If so, unlock the back using `node unlockBatch.js <batch ID>`, delete the processed file marker with `node processedFiles.js -d <filename>`, and then re-store the file in S3"
```

I had tried doing the unlockBatch, but the errors refer to numerous batchIDs, so I didn't make any headway there.  Also I looked at all of the batches and none were locked.

Some minor items for the backlog: 
The error message: `Unable to write usan-cdrlambda-stage/ctg/header/ctg_redshift_header_790047888.csv` is misleading because I assume its not really trying to write to that file, but probably trying to write that string to somewhere.
Also the error message says to ```node unlockBatch.js <batch ID>```.  That syntax is no longer valid - it also needs s3prefix and region.  It looks like you have all the information to put the actual required unlockBatch and processedFiles commands in the error message.  That would be helpful because the paramters of those and other utility functions are inconsistent. For example some require `BatchID` as a param, some require `Batch`, etc.

Cheers,
Jeff
